chaincode-labs/chaincode-residency/2019-06-25-fabrice-drouin-fee-management.md

```diff
@@ -0,0 +1,125 @@
+
+Audience: [inaudible]
+
+Okay if I'm fundee and if I have first of all I may want a commitment transaction to be confirmed quickly because I don't want to have to wait days or weeks before it's suspended as well for me. Also ... I think that's the only reason actually. I can't I can't see why you would need to care about the HTLC second stage (?) transactions now so I think the only problem is you want to be able to have the output that is for you usable as soon as you can and there is no way for you to bump its fees. So unless we add these opt to (?) outputs, that is a bit hacky, there is no way to bump the fees on the commitment transactions. if we had the opt to (?_ outputs then anyone could spend it could be even even miners could spend it that was the original idea the miners would probably want you to clean this up or just anyone now there is there is no simple way of being the fees of a commitment transaction is being either are too low and you can’t use RBF you can’t use cpfv and basically you’re stuck. 
+
+Fabrice: If the fees are too high it's not much better anyway
+
+Audience: [inaudible]
+
+Fabrice: If if with sighhash_anyonecanpay a sighash_single then you don't need these you don't need to worry about fees for these transactions. You could have the zero fees on these transactions and when you want to spend them you can add any input and output you want
+
+Audience: [inaudible]
+
+Fabrice: Yes, because it's one input one output so sighash_single combined with sighash_anyonecanpay is a nice fits but it doesn't work for the commitment transaction
+
+Audience: [inaudible]
+
+Fabrice: No because this has to be confirmed first
+
+Audience: [inaudible]
+
+Fabrice: Yes it's the yes that's the same delay, I basically it's the delay that Bob needs to punish Alice if she tries to cheat so this is the same delay
+
+Audience: [inaudible]
+
+Fabrice: Something that is these are outputs and there's one input in the commitment transaction, these are all outputs I mean it's not it's a bit confusing on the diagram but…
+
+Audience: [inaudible]
+
+Fabrice: That's one of the reasons why we don't have dual funded channels in lightning because it makes a lot of things much more complex. So now it's really easy for the dundee transaction pays for everything and pays the fees. If you have dual funding channels how do you split the fees between funder and fundee as I think we said yesterday that it's easy to game and if all the money is moved to the other side does it still make sense - I mean splitting the fees is very difficult and there are also problems on you need to an interactive way of building the funding transaction between between Alice and Bob, each canal includes an output in its design and agree you leak inputs it's it's it's really messy so it was too complex to get right for the first version of lighting, there's a proposal that is open today but it's it's it's really really hard to get.
```

Comment:
```suggestion
Fabrice: That's one of the reasons why we don't have dual funded channels in lightning because it makes a lot of things much more complex. So now it's really easy for the fundee transaction pays for everything and pays the fees. If you have dual funding channels how do you split the fees between funder and fundee as I think we said yesterday that it's easy to game and if all the money is moved to the other side does it still make sense - I mean splitting the fees is very difficult and there are also problems on you need to an interactive way of building the funding transaction between Alice and Bob, each canal includes an output in its design and agree you leak inputs it's really messy so it was too complex to get right for the first version of lighting, there's a proposal that is open today but it's really hard to get.
```